adaptive/utils.py

```diff
@@ -67,3 +68,18 @@ def decorator(method):
         return functools.wraps(other)(method)
 
     return decorator
+
+
+class _RequireAttrsABCMeta(abc.ABCMeta):
+    def __call__(self, *args, **kwargs):
+        obj = super().__call__(*args, **kwargs)
+        for name, type_ in obj.__annotations__.items():
+            try:
+                x = getattr(obj, name)
+                if not isinstance(x, type_):
+                    raise TypeError(
+                        f"The attribute '{name}' is of {type_} instead of {type(x)}."
+                    )
+            except AttributeError as e:
+                raise e(f"Required attribute {name} not set in __init__.")
```

Comment:
```
            try:
                x = getattr(obj, name)
            except AttributeError:
                raise AttributeError(f"Required attribute {name} not set in __init__.") from None
            else:
                if not isinstance(x, type_):
                    raise TypeError(
                        f"The attribute '{name}' is of {type_} instead of {type(x)}."
                    )
```

Reply:
your
```
except AttributeError as e:
    raise e("...")
```
will not work; you can't call an instance of an exception like a function

Reply:
Oops...